components/scream/src/share/atm_process/atmosphere_process.hpp

```diff
@@ -335,6 +346,8 @@ class AtmosphereProcess : public ekat::enable_shared_from_this<AtmosphereProcess
   virtual void set_required_field_impl (const Field<const Real>& f) = 0;
   virtual void set_computed_field_impl (const Field<      Real>& f) = 0;
 
+  virtual void check_required_fields_impl () = 0;
```

Comment:
Are these always required for every subclass? You can give them an empty body here by default if you don't think these checks apply universally. Of course, that invites maintainers not to implement them, so maybe this is better.

Reply:
ahhh, so if i replace this line with ` virtual void check_required_fields_impl ();` then they won't be required to be defined by every subclass?  I think that is probably a better approach.  Most people will use existing process interfaces as templates, which will all have these defined, so I think the risk of a process skipping is low.

Reply:
You also need to give an implementation, so
```
virtual void check_required_fields_impl () {}
```

Reply:
Great! To do this, try

```
virtual void check_required_fields_impl() {}
```

to make sure it has an empty body.

Reply:
Also, if these checks are not allowed to change the state of an atmosphere process, make sure they're declared  with `const` at the end, e.g.

```
virtual void check_required_fields_impl() const {}
```

This rules out the checking as a source of error (unless you want to allow field checks to repair the fields in situ).

Reply:
Would the `repair` feature of the field property checks not work then?

Reply:
nvmd, you said that directly :-)

Reply:
This is a good suggestion, the `check_required_fields` should probably not be allowed to "repair".  But `check_computed_fields` might.